### DISCUSSION TITLE
Generate branches dynamically before the upgrade and set baseBranches

### DISCUSSION THREAD

**Discussion by lmilbaum** [renovatebot/renovate#24303] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Renovate to kick pre upgrade task which creates new branches in the repo according to a specific logic and then set those branches as baseBranches

### What you tried so far.

Nothing much yet.  Is there a way to implement it?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Could you describe the use case in more detail? i.e. step back up a level and describe what you're "really" trying to achieve

**Reply by lmilbaum** [reply]
Sure. We have an upstream repo which we want to mirror to a downstream repo while manipulating the source tree. One of the manipulation is distributing the code between several branches. The set of branches is computed on the fly.
Hence, executing a pre upgrade task to calculate the branches list and setting it as the baseBranches.

**Reply by rarkins** [reply]
I think you'd need to do that in the config.js somehow, Eg using git ls-remote

**Reply by lmilbaum** [reply]
Is it possible to define a pre upgrade script?

**Reply by rarkins** [reply]
There's a feature request for a dynamic renovate.js which could probably achieve what you want
